### PR TITLE
VNET Rules API spec for Postgres and MySQL

### PR DESCRIPTION
**Pull request by rahulgouthamDOTgs**
Adding API Spec for managing VNET rules in Postgres and MySQL .

### CODE REVIEW

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
     "title": "PostgreSQLManagementClient",
-    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure PostgreSQL resources including servers, databases, firewall rules, log files and configurations.",
+    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure PostgreSQL resources including servers, databases, firewall rules, VNET rules, log files and configurations.",
```
Comment (jianghaolu) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json]: This JSON is also invalid: SyntaxError: Unexpected token ] in JSON at position 14605
    at Object.parseJson (/home/travis/build/Azure/azure-rest-api-specs/node_modules/oav/lib/util/utils.js:78:15)
    at SpecValidator.initialize (/home/travis/build/Azure/azure-rest-api-specs/node_modules/oav/lib/validators/specValidator.js:91:20)
    at Object.validateSpec (/home/travis/build/Azure/azure-rest-api-specs/node_modules/oav/lib/validate.js:60:20)
    at Context.<anonymous> (/home/travis/build/Azure/azure-rest-api-specs/test/semantic.js:19:44)
    at callFn (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runnable.js:348:21)
    at Test.Runnable.run (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runnable.js:340:7)
    at Runner.runTest (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:443:10)
    at /home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:549:12
    at next (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:361:14)
    at /home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:371:7
    at next (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:295:14)
    at Immediate.<anonymous> (/home/travis/build/Azure/azure-rest-api-specs/node_modules/mocha/lib/runner.js:339:5)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json [hunk 2]
```diff
@@ -432,6 +432,193 @@
...
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The requested virtual Network Rule Resource state.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/VirtualNetworkRule"
+            }
+          },
```
Comment (jianghaolu) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json]: remove ',' - bad JSON format

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Comment by jianghaolu** [comment]
Adding @ravbhatnagar if this hasn't gone through ARM review.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json (hunk 1 above)
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json]: fixed

**Review by jianghaolu — APPROVED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json [hunk 3]
```diff
@@ -1267,6 +1454,74 @@
...
+      "properties": {
+        "virtualNetworkSubnetId": {
+          "description": "The ARM resource id of the virtual network subnet.",
+          "type": "string"
+        },
+        "ignoreMissingVnetServiceEndpoint": {
+          "description": "Create firewall rule before the virtual network has vnet service endpoint enabled.",
+          "type": "boolean"
+        },
+        "state": {
```
Comment (ravbhatnagar) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json]: Should this be called "provisioningState"? Are these the states which this resource can go into during the provisioning?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json [hunk 4]
```diff
@@ -432,6 +432,193 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted the virtual network rule."
+          },
+          "default": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json]: default should capture the error response and comply with RPC. Ex here  - https://github.com/Azure/azure-rest-api-specs/blob/current/specification/batch/resource-manager/Microsoft.Batch/2017-05-01/BatchManagement.json#L101

**Review by ravbhatnagar — COMMENTED** [review]
Couple of minor things. Else looks good. 
@rahulgouthamDOTgs - please take a look.

**Comment by ravbhatnagar** [comment]
@jianghaolu - Feel free to merge based on the response to the above questions.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json (hunk 4 above)
Comment (jianghaolu) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json]: Also be aware that `default` is a reserved word in .NET so this property name will be changed in .NET.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json (hunk 3 above)
Comment (rahulgouthamDOTgs) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json]: We are keeping this consistent with SQL APIs , yes a specific VNET rule can be  in one of these states .

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json (hunk 4 above)
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-04-30-preview/postgresql.json]: Sure , we will enhance this eventually , Currently it is capturing all the error responses in the description .

**Pull request merged**